myapp/models.py

```diff
@@ -0,0 +1,270 @@
+                           back_populates="classes")
+
+    def __repr__(self):
+        tpl = ('Class<id: {id}, time: {time}, name: {name}, users: {users},'
+               'labs: {labs}>')
+        formatted = tpl.format(id=self.id, time=self.time,
+                               name=self.name,
+                               users=[u.name for u in self.users],
+                               labs=[lab.name for lab in self.labs])
+        return formatted
+
+
+class User(UserMixin, db.Model):
+    __tablename__ = 'User'
+    id = db.Column(db.String(64), nullable=False, unique=True,
+                   primary_key=True)
+    username = db.Column(db.String(64), nullable=False, unique=True)
+    password = db.Column(db.String(128), nullable=False)
+    name = db.Column(db.String(128))
+    # Many-to-One: A role can have multiple users
+    role_name = db.Column(db.String(64), db.ForeignKey('Role.name'))
+    role = db.relationship("Role", back_populates="users")
+
+    # Many-to-Many: A User can have multiple classes
+    classes = db.relationship("Class", secondary=association_table_class_user,
+                              back_populates="users")
+    # Many-to-Many: A User can have multiple labs
+    labs = db.relationship("Lab", secondary=association_table_user_lab,
+                           back_populates="users")
+
+    def verify_password(self, password):
+        return self.password == password
+
+    # can function checks if user is allowed to peform an operation
+    def can(self, permissions):
+        return self.role is not None and (
+            self.role.permissions & permissions) == permissions
+
+    def __repr__(self):
+        tpl = ('User<id: {id}, username: {username}, password: {password},'
+               ' role: {role_name}, classes: {classes}, labs: {labs}>')
+        formatted = tpl.format(id=self.id, username=self.username,
+                               password=self.password,
+                               role_name=self.role_name,
+                               classes=[c.id for c in self.classes],
+                               labs=[l.id for l in self.labs])
+        return formatted
+
+
+# Function for use when user is not logged in
+class AnonymousUser(AnonymousUserMixin):
+    def can(self, permissions):
+        return False
+# Associate the LoginManager with the anonymous user class
+login_manager.anonymous_user = AnonymousUser
+
+
+# Function accepts a user id and returns the user object of that id
+@login_manager.user_loader
+def load_user(user_id):
+    return User.query.get(user_id)
+
+
+class Role(db.Model):
+    __tablename__ = 'Role'
+    name = db.Column(db.String(64), nullable=False, unique=True,
+                     primary_key=True)
+    # Whether current role object is default of user
+    default = db.Column(db.Boolean, default=False)
+    # A number representing the power the role has using bit operation
+    permissions = db.Column(db.Integer, nullable=False)
```

Comment:
I think that this will make you hate yourself in the future, because it will be difficult to read or print the data in this field without using the methods here, which you may want to do to eliminate the scenario where you inserted incorrect data into the database. You are probably better off storing a list of strings, or if you insist on using numeric values for this field, a list of numbers.